src/test/java/com/teragrep/rlp_03/client/StuckClientCloseTest.java

```diff
@@ -0,0 +1,194 @@
+        ClientFactory clientFactory = new ClientFactory(connectContextFactory, runnableEventLoop.eventLoop());
+
+        try (Client client = clientFactory.open(new InetSocketAddress("localhost", port), 1, TimeUnit.SECONDS)) {
+
+            // send open
+            CompletableFuture<RelpFrame> open = client
+                    .transmit("open", "open be stuck".getBytes(StandardCharsets.UTF_8));
+
+            // send syslog
+            CompletableFuture<RelpFrame> syslog = client
+                    .transmit("syslog", "this syslog is not processed either ".getBytes(StandardCharsets.UTF_8));
+
+            // send close
+            CompletableFuture<RelpFrame> close = client.transmit("close", "".getBytes(StandardCharsets.UTF_8));
+
+            // closing the client, now futures should complete exceptionally
+            client.close();
+
+            AtomicLong completedTransactions = new AtomicLong();
+            // test open response
+            Assertions.assertTrue(open.isCompletedExceptionally());
+            open.whenComplete((relpFrame, throwable) -> {
+                Assertions
+                        .assertEquals(
+                                "TransactionService closed before transaction was completed.", throwable.getMessage()
+                        );
+                Assertions.assertInstanceOf(TransactionServiceClosedException.class, throwable);
+                completedTransactions.getAndIncrement();
+            });
+
+            // test syslog response
+            Assertions.assertTrue(syslog.isCompletedExceptionally());
+            syslog.whenComplete((relpFrame, throwable) -> {
+                Assertions
+                        .assertEquals(
+                                "TransactionService closed before transaction was completed.", throwable.getMessage()
+                        );
+                Assertions.assertInstanceOf(TransactionServiceClosedException.class, throwable);
+                completedTransactions.getAndIncrement();
+            });
+
+            // test close response
+            Assertions.assertTrue(close.isCompletedExceptionally());
+            close.whenComplete((relpFrame, throwable) -> {
+                Assertions
+                        .assertEquals(
+                                "TransactionService closed before transaction was completed.", throwable.getMessage()
+                        );
+                Assertions.assertInstanceOf(TransactionServiceClosedException.class, throwable);
+                completedTransactions.getAndIncrement();
+            });
+
+            Assertions.assertEquals(3, completedTransactions.get());
+
+            // close the client eventLoop
+            runnableEventLoop.close();
+            eventLoopThread.join();
+        }
+        catch (InterruptedException | ExecutionException | IOException | TimeoutException exception) {
+            throw new RuntimeException(exception);
```

Comment:
I'd Assertions.assertDoesntThrow all that can be asserted in the try block and Assertions.fail("Message") in the catch block